numeric.c

```diff
@@ -1820,6 +1820,25 @@ rb_flo_is_finite_p(VALUE num)
     return Qtrue;
 }
 
+enum flo_prevnext_flags
+{
+    FLOAT_PREV = 0,
+    FLOAT_NEXT = 1
+};
+
+static VALUE
+flo_prev_or_next(VALUE flo, int flag)
+{
+    double x, y;
+    x = NUM2DBL(flo);
+    if (flag == 1)
```

Comment:
You can use your own flag here: ``if (flag == FLOAT_NEXT)``.
You could also use HUGE_VAL instead of a new flag:
``return flo_prev_or_next(vx, HUGE_VAL);`` and ``return flo_prev_or_next(vx, -HUGE_VAL);``